pkg/controller/dataimportcron-controller.go

```diff
@@ -394,13 +394,14 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 		return res, err
 	}
 
-	// We use the poller returned reconcile.Result for RequeueAfter if needed
-	// skip if we disabled schedule
+	// Skip if schedule is disabled
 	if isImageStreamSource(dataImportCron) && dataImportCron.Spec.Schedule != "" {
-		res, err := r.pollImageStreamDigest(ctx, dataImportCron)
+		// We use the poll returned reconcile.Result for RequeueAfter if needed
+		pollRes, err := r.pollImageStreamDigest(ctx, dataImportCron)
```

Comment:
Would it be simpler to do
```go
res, err = r.pollImageStreamDigest(ctx, dataImportCron)
if err != nil {
  return res, err
}
```
Note the subtle change from `:=` to `=` which essentially assigns res to the result of pollImageStreamDigest. Which is logically the same as all the code here.

Reply:
This was the original solution but my concern was that we'd trip over this again
https://github.com/kubevirt/containerized-data-importer/pull/2823#discussion_r1276461398

Reply:
Apparently this was the original solution, but not explicit enough with regards to how res and err are assigned. I have no issues with either approach honestly.